internal/framework/status/updater.go

```diff
@@ -250,6 +254,24 @@ func (upd *UpdaterImpl) writeStatuses(
 	}
 }
 
+// UpdateAddresses is called when the Gateway Status needs its addresses updated.
+func (upd *UpdaterImpl) UpdateAddresses(ctx context.Context, svc *v1.Service) {
+	addresses, err := GetGatewayAddresses(ctx, upd.cfg.Client, svc, upd.cfg.GatewayPodConfig)
+	if err != nil {
+		upd.cfg.Logger.Error(err, "Setting GatewayStatusAddress to Pod IP Address")
+	}
+
+	for name, status := range upd.lastStatuses.gatewayAPI.GatewayStatuses {
```

Comment:
Michael had a comment above about moving the `getGatewayAddresses` call to the handler, which I think makes sense. However, the reason I don't call the normal Update() method from the handler is because it needs a graph to passto the Update() call. We aren't rebuilding the entire graph on an address update, so that's why I have this separate function to get the last statuses.